crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -756,24 +767,18 @@ where
 
         match expr {
             ast::Expr::Name(name_node @ ast::ExprName { id, ctx, .. }) => {
-                let flags = match (ctx, self.current_assignment) {
+                let (is_use, is_definition) = match (ctx, self.current_assignment) {
                     (ast::ExprContext::Store, Some(CurrentAssignment::AugAssign(_))) => {
                         // For augmented assignment, the target expression is also used.
-                        SymbolFlags::IS_DEFINED | SymbolFlags::IS_USED
-                    }
-                    (ast::ExprContext::Store, Some(CurrentAssignment::AnnAssign(ann_assign)))
-                        if ann_assign.value.is_none() =>
-                    {
-                        // An annotated assignment that doesn't assign a value is not a Definition
-                        SymbolFlags::empty()
+                        (true, true)
                     }
-                    (ast::ExprContext::Load, _) => SymbolFlags::IS_USED,
-                    (ast::ExprContext::Store, _) => SymbolFlags::IS_DEFINED,
-                    (ast::ExprContext::Del, _) => SymbolFlags::IS_DEFINED,
-                    (ast::ExprContext::Invalid, _) => SymbolFlags::empty(),
+                    (ast::ExprContext::Load, _) => (true, false),
+                    (ast::ExprContext::Store, _) => (false, true),
+                    (ast::ExprContext::Del, _) => (false, true),
+                    (ast::ExprContext::Invalid, _) => (false, false),
```

Comment:
the previous approach that used bitflags here seemed slightly more elegant to me :/

Reply:
We could switch back to bitflags here, but there's more context, and a couple downsides in my mind:

1) I think it's confusing to use the `SymbolFlags` bitflags here, since we don't actually use them to create the symbol with flags anymore (the bitflags aren't even directly exposed outside the symbol table anymore). Setting the "bound" flag is now handled automatically in `add_definition`, which I think is definitely an improvement, since it reduces duplication and the possibility of things getting out of sync due to a mistake in the builder. And defining a set of two bitflags just for use in this method feels not worth it.

2) Using a bitflag here is perhaps a bit clearer, but it makes checking the value later significantly more verbose; I'm not really convinced it's a net gain.